docs/specification/README.md

```diff
@@ -166,7 +166,7 @@ Channel Info records are uniquely identified within a file by their channel ID.
 | 2 | id | uint16 | A unique identifier for this channel within the file. |
 | 4 + N | topic | String | The channel topic. |
 | 4 + N | message_encoding | String | Encoding for messages on this channel. The value should be one of the [well-known message encodings](./well-known-encodings.md). Custom values should use `x-` prefix. |
-| 2 | schema_id | uint16 | The schema for messages on this channel. |
```

Comment:
Is this contentious? Seems helpful for `json` or `h264` encoded channels.

Reply:
It's nice to provide a name for the datatype even if you don't need a schema. I don't think this change belongs in this PR.

Reply:
Nice if you’re using ros 2 and just don’t have the schema data. But for json or h264 why do I have to make up a schema name? That’s silly.

I can split PR if you preferred.

Reply:
If you are using ros2 and don't have schema data mcap is not for you. I think we are getting a bit far from the plot here. Mcap is about storing schematized messages. If you want to record opaque bytes is mcap really the format for that?

Reply:
I didn't say anything about opaque bytes. If I add an h264 NAL stream to an mcap file (a use case we have previously discussed as valid) there is no schema or schema name, only a message encoding. As far as I can tell the only solutions I can see to this are:

- h264 or json streams with no schema are not use cases we want to support
- Create an empty Schema with no name, schema encoding, or data
- Alllow a "null" (zero byte) schema_id

Reply:
> If you are using ros2 and don't have schema data mcap is not for you.

If we want to support converting .db3 to .mcap, there are absolutely going to be some files generated that have a schema name but no schema data. However, that's unrelated to this change because you would still need a Schema record in that instance to save the schema name. I'm pushing the discussion above about what to do if you have no schema name either, just a stream of self-describing encoded data.